src/targets/services/checkAndUpdate.js

```diff
@@ -0,0 +1,95 @@
+import CozyClient from 'cozy-client'
+import {
+  schema,
+  DOCTYPE_CONTACTS_VERSION,
+  DOCTYPE_CONTACTS
+} from '../../helpers/doctypes'
+import log from 'cozy-logger'
+
+const client = CozyClient.fromEnv(null, { schema })
+
+export const getContactsToUpdate = async () => {
+  const queryDef = client
+    .find(DOCTYPE_CONTACTS)
+    .where({
+      trashed: {
+        $exists: false
+      },
+      $or: [
+        {
+          'cozyMetadata.doctypeVersion': { $exists: false }
+        },
+        {
+          'cozyMetadata.doctypeVersion': { $lt: DOCTYPE_CONTACTS_VERSION }
+        }
+      ]
+    })
+    .indexFields(['_id'])
+    .limitBy(1000)
+
+  try {
+    const data = await client.queryAll(queryDef)
+    log('info', `${Object.keys(data).length} contact(s) found`)
+    return data
+  } catch (e) {
+    log('error', `Can't find elements : ${e}`)
+    return
```

Comment:
comme tu throw pas l'erreur, si on a une erreur, on va faire ici : 
https://github.com/cozy/cozy-contacts/pull/658/files#diff-7537a1c17a07ccf377a4e610977feaffR82

undefined.map() => erreur et ça va afficher "contact not updated" alors que c'est pas vraiment ce qui s'est passé